internal/template/template.go

```diff
@@ -0,0 +1,51 @@
+package template
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+var archiveFolderName = "files"
+var tarName = archiveFolderName + ".tar.gz"
+
+// GetTemplateFiles returns the path to a directory that contains the intial
+// project template.
+func GetTemplateFiles(ctx context.Context, templateURL string) (string, error) {
+	dir, err := os.MkdirTemp("", "exo-template-clone-")
+	if err != nil {
+		return "", fmt.Errorf("making temp dir: %w", err)
+	}
+
+	url := fmt.Sprintf("%s/%s", templateURL, tarName)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", fmt.Errorf("getting template files: %w", err)
+	}
+	defer resp.Body.Close()
+
+	err = uncompress(resp.Body, dir)
+	if err != nil {
+		return "", fmt.Errorf("untarring template: %w", err)
+	}
+	return filepath.Join(dir, archiveFolderName), nil
+}
+
+// MakeTemplateFiles takes a directory of files to template and outputs the
+// resultant template into the given directory.
+func MakeTemplateFiles(ctx context.Context, inputDir, outputDir string) error {
```

Comment:
Is this actually make-template-_file_ singular?

Reply:
My thinking was that the template structure is a directory. It so happens that right now the directory only contains one file (a `files.tar.gz`) but the expectation is that it will contain more for metadata like icons or descriptions of the template.

Reply:
gotcha - we should probably document the template format somewhere. maybe in a README or `go.doc` file or something. A json schema or typescript signature for the metadata is probably a good idea once we get there.